src/main/java/org/gridsuite/study/server/service/dynamicsimulation/impl/DynamicSimulationServiceImpl.java

```diff
@@ -114,28 +122,43 @@ public List<DoubleTimeSeries> getTimeSeriesResult(UUID nodeUuid, List<String> ti
     }
 
     @Override
-    public List<StringTimeSeries> getTimeLineResult(UUID nodeUuid) {
-        List<TimeSeries> timeLines = new ArrayList<>();
+    public List<TimeLineEventInfos> getTimeLineResult(UUID nodeUuid) {
+        List<TimeLineEventInfos> timeLineEvents = new ArrayList<>();
 
         Optional<UUID> resultUuidOpt = networkModificationTreeService.getComputationResultUuid(nodeUuid, ComputationType.DYNAMIC_SIMULATION);
 
         if (resultUuidOpt.isPresent()) {
             UUID timeLineUuid = dynamicSimulationClient.getTimeLineResult(resultUuidOpt.get()); // get timeline uuid
             if (timeLineUuid != null) {
                 // get timeline data
-                timeLines = timeSeriesClient.getTimeSeriesGroup(timeLineUuid, null);
+                List<TimeSeries> timeLines = timeSeriesClient.getTimeSeriesGroup(timeLineUuid, null);
 
                 // get first element to check type
                 if (!CollectionUtils.isEmpty(timeLines) &&
                     !(timeLines.get(0) instanceof StringTimeSeries)) {
                     throw new StudyException(StudyException.Type.TIME_SERIES_BAD_TYPE, "Time lines can not be a type: "
-                       + timeLines.get(0).getClass().getSimpleName()
-                       + ", expected type: " + StringTimeSeries.class.getSimpleName());
+                                                                                       + timeLines.get(0).getClass().getSimpleName()
+                                                                                       + ", expected type: " + StringTimeSeries.class.getSimpleName());
+                }
+
+                // convert {@link StringTimeSeries} to {@link TimeLineEventInfos}
+                // note that each {@link StringTimeSeries} corresponds to an array of {@link TimeLineEventInfos}
+                if (!CollectionUtils.isEmpty(timeLines)) {
+                    timeLines.stream()
+                            .flatMap(series -> Stream.of(((StringTimeSeries) series).toArray()))
+                            .forEach(eventJson -> {
```

Comment:
This should be done in a separate for loop.

Reply:
https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwiZrKWHjdGEAxXNe6QEHbHdDLwQFnoECBoQAQ&url=https%3A%2F%2Fstackoverflow.com%2Fquestions%2F16635398%2Fjava-8-iterable-foreach-vs-foreach-loop&usg=AOvVaw2hFOxnIJLeenz0dTQxNMYZ&opi=89978449

Reply:
I use _map_ operator instead of _forEach_ to avoid side-effects
DONE! by https://github.com/gridsuite/study-server/pull/532/commits/3e13c3da0cc83d780be39899dd063bb7f2c27c36